classes/class-p4-master-site.php

```diff
@@ -605,6 +614,59 @@ public function view_meta_box_search( $post ) {
 		<?php
 	}
 
+	/**
+	 * Forces a user to login using Google Auth if they have a greenpeace.org email
+	 *
+	 * @param WP_User $user The current user logging in.
+	 * @param String  $username The username of the user.
+	 * @param String  $password The password of the user.
+	 */
+	public function enforce_google_signon( $user, $username = null, $password = null ) {
+
+		if ( empty( $username ) || empty( $password ) ) {
+			return $user;
+		} elseif ( strpos( $username, '@' ) ) {
+			$user_data = get_user_by( 'email', trim( wp_unslash( $username ) ) );
+			if ( empty( $user_data ) ) {
+				return $user;
+			}
+		} else {
+			$login     = trim( $username );
+			$user_data = get_user_by( 'login', $login );
+		}
+
+		if ( defined( 'WP_DEBUG' ) && WP_DEBUG === true ) {
+			return $user;
+		}
+
+		if ( is_wp_error( $user ) ) {
+			return $user;
+		}
+
+		$domain = '@greenpeace.org';
+		if ( mb_substr( $user_data->data->user_email, -strlen( $domain ) ) === $domain ) {
```

Comment:
I reach this point if I try to log in with a user that does not exist, which triggers some notices because `$user_data` is `false`, so to not have notices and because `get_user_by()` returns `WP_User|false` I would change the first `if` like this
```php
if ( empty( $username ) || empty( $password ) ) {
	return $user;
} 

if ( strpos( $username, '@' ) ) {
	$user_data = get_user_by( 'email', trim( wp_unslash( $username ) ) );
} else {
	$login     = trim( $username );
	$user_data = get_user_by( 'login', $login );
}

if ( false === $user_data ) {
	return $user;
}
```

Reply:
Good catch!